tools/upgradetool/src/main/java/org/openhab/core/tools/UpgradeTool.java

```diff
@@ -58,42 +83,113 @@ public static void main(String[] args) {
             String loglevel = commandLine.hasOption(OPT_LOG) ? commandLine.getOptionValue(OPT_LOG).toUpperCase()
                     : "INFO";
             if (!LOG_LEVELS.contains(loglevel)) {
-                System.out.println("Allowed log-levels are " + LOG_LEVELS);
+                println("Allowed log-levels are " + LOG_LEVELS);
+                System.exit(0);
+            }
+
+            if (commandLine.hasOption(OPT_LIST_COMMANDS)) {
+                println("Available commands:");
+                UPGRADERS.stream().forEach(upgrader -> {
+                    println(" - " + upgrader.getName() + ": " + upgrader.getDescription());
+                });
                 System.exit(0);
             }
 
             System.setProperty(org.slf4j.simple.SimpleLogger.DEFAULT_LOG_LEVEL_KEY, loglevel);
 
-            String baseDir = commandLine.hasOption(OPT_DIR) ? commandLine.getOptionValue(OPT_DIR)
+            boolean force = commandLine.hasOption(OPT_FORCE);
+            String command = commandLine.hasOption(OPT_COMMAND) ? commandLine.getOptionValue(OPT_COMMAND) : null;
+
+            if (command != null && UPGRADERS.stream().filter(u -> u.getName().equals(command)).findAny().isEmpty()) {
+                println("Unknown command: " + command);
+                System.exit(0);
+            }
+
+            String userdataDir = commandLine.hasOption(OPT_USERDATA_DIR) ? commandLine.getOptionValue(OPT_USERDATA_DIR)
                     : System.getenv("OPENHAB_USERDATA");
-            if (baseDir == null || baseDir.isBlank()) {
-                System.out.println(
-                        "Please either set the environment variable ${OPENHAB_USERDATA} or provide a directory through the --dir option.");
+            if (command == null && (userdataDir == null || userdataDir.isBlank())) {
```

Comment:
I believe userdata folder is needed for all other commands than tags conversion.

Reply:
Obsolete comment